observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/deployment/OpenTelemetryDependencyProcessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.opentelemetry.deployment;
+
+import static org.wildfly.extension.opentelemetry.OpenTelemetrySubsystemDefinition.EXPORTED_MODULES;
+import static org.wildfly.extension.opentelemetry.OpenTelemetrySubsystemDefinition.MODULES;
+
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.server.deployment.module.ModuleDependency;
+import org.jboss.as.server.deployment.module.ModuleSpecification;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleLoader;
+
+public class OpenTelemetryDependencyProcessor implements DeploymentUnitProcessor {
+    @Override
+    public void deploy(DeploymentPhaseContext phaseContext) {
+        addDependencies(phaseContext.getDeploymentUnit());
+    }
+
+    @Override
+    public void undeploy(DeploymentUnit deploymentUnit) {
+
+    }
+
+    private void addDependencies(DeploymentUnit deploymentUnit) {
+        ModuleSpecification moduleSpecification = deploymentUnit.getAttachment(Attachments.MODULE_SPECIFICATION);
+        ModuleLoader moduleLoader = Module.getBootModuleLoader();
+        for (String module : MODULES) {
+            moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, module, false, false,
+                    true, false));
+        }
```

Comment:
Why does the OpenTelemetry subsystem module need to be exposed to deployments?
Unless I'm mistaken, it seems to me that this is only needed to allow Weld to locate the CDI extension.
Would it not be better to register this extension manually (i.e. via WeldCapability#registerExtensionInstance(...))?
This way we avoid bloating the deployment classloader and do not expose unnecessary subsystem internals to deployments.

Reply:
To be honest, I was just following the pattern I had seen elsewhere. :) I'll take a look at the linked code and go from there.

Reply:
An alternative approach, leveraged in subsystems like messaging-activemq or smallrye-fault-tolerance: encapsulate your portable extension within a separate module, and add a deployment dependency on that module.

Reply:
Ideally, the only modules on which a deployment should depend are any public modules containing API (spec or otherwise) that a deployment intends to use.
If you register your portable extension via the WeldCapability directly, there should be no need to export that module to the deployment (since Weld no longer needs to resort to service loading to find your CDI extension).

Reply:
Fixed.